limitador/src/storage/infinispan/counters.rs

```diff
@@ -0,0 +1,124 @@
+// Infinispan does not support counters with TTL. This module provides some
+// functions on top of the Infinispan client to be able to work with counters
+// with TTLs.
+//
+// Counters in Infinispan don't have a TTL. So to implement a counter, apart
+// from the actual counter, we need a helper key with a TTL that simply tells us
+// whether the counter is still valid.
+//
+// This is how it works in more detail: When a counter is created, apart from
+// the actual Infinispan counter, this module creates a regular entry with the
+// same name as the counter and with a TTL. The TTL is set to the duration of
+// the limit. So, for a limit of 10 requests per minute, the TTL would be a
+// minute. While the key has not expired, we know that the value of the counter
+// is valid. When the key has expired, the value of the counter no longer
+// applies.
+
+use crate::storage::infinispan::response::response_to_string;
+use infinispan::errors::InfinispanError;
+use infinispan::{request, Infinispan};
+use std::time::Duration;
+
+pub struct CounterOpts {
+    initial_value: i64,
```

Comment:
@unleashed This should be `CounterVal`, but when I was writing this I realized that it's not exposed in `infinispan-rs`.

Reply:
Similar to https://github.com/3scale-labs/limitador/pull/38#discussion_r629225082